src/storage-controller/src/lib.rs

```diff
@@ -1951,16 +2045,9 @@ where
                 updated_frontiers = Some(Response::FrontierUpdates(updates));
             }
             Some(StorageResponse::DroppedId(id)) => {
-                tracing::debug!("DroppedId for collection {id}");
-
-                if let Some(_collection) = self.collections.remove(&id) {
-                    // Nothing to do, we already dropped read holds in
-                    // `drop_sources_unvalidated`.
-                } else {
-                    soft_panic_or_log!(
-                        "DroppedId for ID {id} but we have neither ingestion nor export \
-                         under that ID"
-                    );
+                let prev = self.dropped_ids.remove(&id);
```

Comment:
How does this interact with multi-replica storage clusters? Do we only expect to get one dropped id message globally?

Reply:
Ah yeah, dammit, `DroppedId` is a bit confusing, because for some things I thing the replica client logic only forwards a thing when all of the replicas agree, but I think you're right and for `DroppedId` they pass through as is, so we get multiple of them. I would just not remove things from `dropped_ids` and accept the small cost of keeping these around until a restart. ((I don't like using `dropped_ids` to begin with, but it's a compromise. ))

Reply:
> I would just not remove things from dropped_ids and accept the small cost of keeping these around until a restart.

I'm not a fan of intentionally leaking memory. I think the storage controller has done that previously with the collection (and/or export) state and it turned out fine, but someone running `CREATE`/`DROP` in a loop could still oom envd. We definitely can't do it in compute where transient dataflows are created and dropped all the time.